### DISCUSSION TITLE
Ping servers after instance being down

### DISCUSSION THREAD

**Discussion by NoelDeMartin** [mastodon/mastodon#16722] [Q&A] [answered]
Hi there,

I recently went on holidays and my instance when down for a while (I'm not sure how long, probably a week or so). When I came back and fixed it, I've noticed that I stopped getting toots from some accounts. I guess they marked my instance as dead given the inactivity.

Now, it seems like interacting with those instances fixed it. For example, I boosted a toot from `fosstodon.org` and now I'm getting toots from accounts in that instance. But the issue persists with other domains. Is there a way to ping all domains to let them now that my instance is back?

**Comment by ClearlyClaire** [comment, marked as answer]
That is a good question. Remote servers are marked as unavailable for delivery if delivering messages to them fails for 7 different days without any success in-between, so that is probably what happened with your instance. Whenever getting a message from your instance, it will be marked as available for delivery again.

Pinging all domains is generally not workable given that the sheer amount of them, and there is currently no built-in way to do that in Mastodon. There is also no “ping” activity in ActivityPub, so what message to send isn't obvious either.

Just posting something or updating your profile should take care of instances where you have followers, but I think that has been resolved already by now.

The next set of instances that is likely to be affected is the instances who have accounts that you follow. If we limit ourselves to that, the query might be reasonable (you can run it in the Rails console by doing `RAILS_ENV=production bundle exec rails c`):

```ruby
# Inboxes of accounts/servers we follow
inboxes = Account.remote.where('EXISTS (SELECT 1 FROM follows WHERE follows.target_account_id = accounts.id)').inboxes.to_a
# Remove inboxes of accounts/servers who follow us
inboxes -= Account.remote.where('EXISTS (SELECT 1 FROM follows WHERE follows.account_id = accounts.id)').inboxes

# Schedule sending a dummy payload to those inboxes. This isn't a valid activity, but it should be enough to reset delivery failure tracker in current Mastodon versions

ActivityPub::DeliveryWorker.push_bulk(inboxes) do |inbox_url|
  ['{}', -99, inbox_url]
end
```

**Reply by NoelDeMartin** [reply]
I did try running `tootctl domains crawl` to see if that did the trick, but that didn't seem to work either. Your solution looks promising, I'll give it a try and let you know how it went.

Thanks for your help :).

**Reply by NoelDeMartin** [reply]
When you posted the answer, I tried updating my profile because I hadn't done that (nor tooted), and that seemed to fix a bunch of domains. Now I just run the code and I didn't get any errors, so I suppose it worked :). I'm not sure because updating the profile already fixed the domains I was aware of, so I can't tell if there were still some pending.

But my problem is fixed, so I'm marking this as the answer, thanks again!

**Comment by cassieroyal** [comment]
I know this is an email from mastodon but are you referring to my VIPgirlfriend account or my tusk account and if this about my VIP girlfriend account when can I post another profile and and how can I prevent spam from happening again.

Sent from Yahoo Mail on Android 
 
  On Fri, Sep 17, 2021 at 2:14 AM, Noel De ***@***.***> wrote:   


When you posted the answer, I tried updating my profile because I hadn't done that (nor tooted), and that seemed to fix a bunch of domains. Now I just run the code and I didn't get any errors, so I suppose it worked :). I'm not sure because updating the profile already fixed the domains I was aware of, so I can't tell if there were still some pending.

But my problem is fixed, so I'm marking this as the answer, thanks again!

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by ClearlyClaire** [reply]
Hi @cassieroyal. Sorry, but this GitHub project is for Mastodon development.

As far as I understand your messages, you are trying to get support with regards to your account on “VIPgirlfriend”. While that website seems to be running (an old version of) Mastodon, we have no relation with it and cannot help you with your query.
